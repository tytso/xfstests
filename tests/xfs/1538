#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-later
# Copyright (c) 2022-2024 Oracle.  All Rights Reserved.
#
# FS QA Test No. 1538
#
# Populate a XFS filesystem and fuzz every rtrefcountbt record field.
# Use xfs_scrub to fix the corruption.

. ./common/preamble
_begin_fstest dangerous_fuzzers dangerous_scrub dangerous_online_repair realtime

_register_cleanup "_cleanup" BUS

. ./common/filter
. ./common/populate
. ./common/fuzzy
. ./common/reflink

_require_realtime
_require_scratch_reflink
_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

path="$(_scratch_xfs_find_rgbtree_height 'refcnt' 2)" || \
	_fail "could not find two-level rtrefcountbt"
inode_ver=$(_scratch_xfs_get_metadata_field "core.version" "path -m $path")

echo "Fuzz rtrefcountbt recs"
_scratch_xfs_fuzz_metadata '' 'online' "path -m $path" "addr u${inode_ver}.rtrefcbt.ptrs[1]" >> $seqres.full
echo "Done fuzzing rtrefcountbt recs"

# success, all done
status=0
exit
