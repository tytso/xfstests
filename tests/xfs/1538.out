QA output created by 1538
Format and populate
Fuzz rtrefcountbt recs
Done fuzzing rtrefcountbt recs
